src/simplelink.rs

```diff
@@ -240,3 +252,165 @@ impl SimpleLink {
         Ok(())
     }
 }
+
+// File System
+//
+
+pub trait Read {
+    fn read(&mut self, buf: &mut[u8]) -> Result<usize, SimpleLinkError>;
+    fn read_to_end(&mut self, buf: &mut Vec<u8>) -> Result<usize, SimpleLinkError>;
+    fn read_to_string(&mut self, str: &mut String) -> Result<usize, SimpleLinkError>;
+}
+
+pub trait Write {
+    fn write(&mut self, buf: &[u8]) -> Result<usize, SimpleLinkError>;
+    fn flush(&mut self) -> Result<(), SimpleLinkError>;
+}
+
+pub trait Seek {
+    fn seek(&mut self, pos: u64) -> Result<u64, SimpleLinkError>;
+}
+
+pub struct File {
+    offset: usize,
+    file_handle: i32,
+}
+
+impl File {
+    /// Returns a file's info structure from the file system
+    pub fn get_info(file_name: &str) -> Result<SlFsFileInfo, SimpleLinkError> {
+        let mut file_info = SlFsFileInfo {
+            flags : 0,
+            file_length : 0,
+            allocated_length : 0,
+            token : [0; 4]
+        };
+        try_fs!(sl_FsGetInfo(file_name.as_ptr(), 0, &mut file_info) as i32);
+        Ok(file_info)
+    }
+
+    /// Removes a file fro the file system
+    pub fn remove(file_name: &str) {
+        unsafe {
+            sl_FsDel(file_name.as_ptr(), 0);
+        }
+    }
+
+    /// Opens a file for reading
+    pub fn open(file_name: &str) -> Result<File, SimpleLinkError> {
+        File::open_with_mode(file_name, File::mode(false, false, 0))
+    }
+
+    /// Opens a file for writing; possibly creating it in the process
+    pub fn create(file_name: &str) -> Result<File, SimpleLinkError> {
+        File::open_with_mode(file_name, File::mode(true, true, 57344))
+    }
+
+    /// Write buffers to file system
+    pub fn sync_all(&self) -> Result<(), SimpleLinkError> {
+        Ok(())
+    }
+
+    /// Write file data to file system
+    pub fn sync_data(&self) -> Result<(), SimpleLinkError> {
+        Ok(())
+    }
```

Comment:
I get that, but I'm not a fan of making it look like something important is implemented while it's not.